src/test/nodestore/DatabaseShard_test.cpp

```diff
@@ -0,0 +1,894 @@
+namespace NodeStore {
+
+// Tests DatabaseShard class
+//
+class DatabaseShard_test : public TestBase
+{
+    static constexpr std::uint32_t max_size_gb = 10;
+    static constexpr std::uint32_t ledgers_per_shard = 256;
+    static constexpr std::uint32_t earliest_seq = ledgers_per_shard + 1;
+    static constexpr std::uint32_t data_size_max = 16;
+    static constexpr std::uint32_t ini_amount = 1000000;
+    static constexpr std::uint32_t n_test_shards = 10;
+    static constexpr std::uint32_t shard_store_timeout = 60;
+    test::SuiteJournal journal_;
+    beast::temp_dir defNodeDir;
+
+    struct test_data
+    {
+        beast::xor_shift_engine rng_;
+        int n_shards_;
+        std::vector<std::shared_ptr<test::jtx::Account> > A_;
+        std::vector<int> acc_;
+        std::vector<std::vector<std::pair<int,int>>> pay_;
+        std::vector<int> xrp_;
+        std::vector<std::shared_ptr<const Ledger>> ledgers_;
+
+        test_data(std::uint64_t const seedValue,
+                  int data_size = data_size_max,
+                  int n_shards = 1)
+            : rng_(seedValue)
+            , n_shards_(n_shards)
+        {
+            std::uint32_t n = 0;
+            for (std::uint32_t i = 0; i < ledgers_per_shard * n_shards; ++i)
+            {
+                int p;
+                if (n >= 2)
+                    p = rand_int(rng_, 2 * data_size);
+                else
+                    p = 0;
+
+                std::vector<std::pair<int,int>> pay;
+
+                for (int j = 0; j < p; ++j)
+                {
+                    int from, to;
+                    do
+                    {
+                        from = rand_int(rng_, n - 1);
+                        to = rand_int(rng_, n - 1);
+                    }
+                    while (from == to);
+
+                    pay.push_back(std::make_pair(from, to));
+                }
+
+                n += !rand_int(rng_, (ledgers_per_shard * n_shards) / data_size);
+
+                if (n > A_.size())
+                {
+                    char str[9];
+                    for (int j = 0; j < 8; ++j)
+                        str[j] = 'a' + rand_int(rng_, 'z' - 'a');
+                    str[8] = 0;
+                    A_.push_back(std::make_shared<test::jtx::Account>(str));
+                }
+
+                acc_.push_back(n);
+                pay_.push_back(pay);
+                xrp_.push_back(rand_int(rng_, 90) + 10);
+            }
+        }
+
+        bool new_acc(int seq)
+        {
+            return acc_[seq] > (seq ? acc_[seq - 1] : 0);
+        }
+
+        void make_ledger_data(test::jtx::Env &env_, std::uint32_t seq)
+        {
+            using namespace test::jtx;
+
+            if (new_acc(seq))
+                 env_.fund(XRP(ini_amount), *A_[acc_[seq] - 1]);
+
+            for (std::uint32_t i = 0; i < pay_[seq].size(); ++i)
+            {
+                env_(pay(*A_[pay_[seq][i].first], *A_[pay_[seq][i].second], XRP(xrp_[seq])));
```

Comment:
Please run this entire file under clang format; the 80 character width limit and other conventions are not being followed.

Reply:
Done